### DISCUSSION TITLE
Things to add to Reco✨

### DISCUSSION THREAD

**Discussion by Arvinth-Krishna** [Arvinth-Krishna/Reco-PC-Server#12] [Ideas]
**Suggest**, **Discuss**, **Add** & **Repeat 😎**

**Comment by Arvinth-Krishna** [comment]
**[Previous ideas, suggestions & discussions](https://github.com/Arvinth-Krishna/Reco-PC-Server/issues/8#issue-1065848848)**

I believe that nothing is impossible. So let's add...

**Current Pending features:**

🟠 **!media now** - shows now(current) playing.
🟠 **Update the mobile app** for a cleaner look and all-new functions integrated and more. (Soon! I will, open-source Reco App).
🟠 **!usage** - show current CPU, RAM, GPU, Disk & Network usage.
🟠 **Uptime** time in Reco Activity area and in Statistics section in !reco.
🟠 **Auto bot status** (bot status' (idle, online, dnd) changes according to if the pc is active or idle (screensaver active or display on)
or the option to set a custom timeout of user inactivity.
🟠 **Auto Update Reco Activity feature** which includes now playing, uptime, and system usage: https://github.com/Arvinth-Krishna/Reco-PC-Server/discussions/12#discussioncomment-2055717

**Comment by BigFlubba** [comment]
Sounds good so far, what about my most recent ideas?

**Reply by Arvinth-Krishna** [reply]
Did I miss something in the **Current Pending features** list?

**Reply by BigFlubba** [reply]
Yah, the rich presence and a few other things

**Reply by Arvinth-Krishna** [reply]
Pls, mention the left-out features.

**Reply by Arvinth-Krishna** [reply]
Rich Presence can't be added to the bots. It's for Users.

**Reply by BigFlubba** [reply]
Well with the playing status yes you can change it.

**Reply by BigFlubba** [reply]
But I would have the option if you would either like uptime or system usage or have the system info auto post or upon command. I would suggest that you can have the option for now playing, uptime, and system usage the option to either have it show upon command usage, auto posting (with interval for uptime and system usage (not really a need to now playing because it should (if auto post is enabled) auto post upon song change), and/or custom status. I hope you get what I mean and it makes sense.

**Reply by Arvinth-Krishna** [reply]
Nice suggestion.

**Reply by BigFlubba** [reply]
Thanks.

**Comment by BigFlubba** [comment]
-storage command (list drives, size of each drive, and size remaining)
-apprestart command (restart an app of your choosing instead of doing !appquitter and !launch)
-displayoff command (turns off the displays (I currently press the power button to shut them off (I have it mapped in the control panel like that) because I have my pc always on and that's how I shut them off rather then waiting for screen timeout to happen) (some ways of making this happen is pressing the sleep button (would need to be configured in the control panel first), issuing a script to shut them off, or something to make that happen)

**Comment by Arvinth-Krishna** [comment]
**Reco PC Server -v8.0** features added:

🟠 !media now - shows now(current) playing.
🟠 Update the mobile app for a cleaner look and all-new functions integrated and more. (Soon! I will, open-source Reco App).
🟠 !usage - show current CPU, RAM, GPU, Disk & Network usage.
✔ Uptime time in Reco Activity area and in Statistics section in !reco.
✔Dynamic Bot Status -- Auto bot status (bot status' (idle, online, dnd) changes according to if the pc is active or idle (screensaver active or display on)
or the option to set a custom timeout of user inactivity.
✔Uptime added in Bot Activity.
🟠 Storage command (list drives, size of each drive, and size remaining)
🟠 Apprestart command
🟠 Displayoff command
✔ Alert feature - Notifies you when user interaction is found in your system.
✔ Rich Presence (RPC) is a feature integrated into Reco, which helps you show custom RPC, Reco RPC, and CPU usage.
✔ https://github.com/Arvinth-Krishna/Reco-PC-Server/issues/13 - Option for bot online notify message to send in channels of choosing.

**[Download Now! (v8.0)](https://github.com/Arvinth-Krishna/Reco-PC-Server/archive/refs/tags/v8.0.zip)**

**Comment by Arvinth-Krishna** [comment]
🌟 **Upcoming Features!**

- **!press**       -> Helps you to use a key or a combination of keys(**Hotkeys**)
- **!type**        -> Helps you write any strings.
- **!file print**  -> This command helps you to print files directly while you are using !file command. (FYI, all printer command's optional arguments can be used with this command.)
- **!startmenu** or **!sm** -> This command will automatically press start, then search and launch the app for you.
- **!alert**        -> We will be introducing **Reco Notifications** role will be assigned automatically when you use command alert. This feature will help u get notification if you have set your server to mentions only(Notification).
- **!usage**     ->  Show current CPU, RAM, GPU, Disk & Network usage.

**Comment by BigFlubba** [comment]
Ability for !cmd or !powershell to have a continuous single instance. What I mean is instead of doing every command in a new instance of cmd or powershell you can have more than one command in one instance.

Ex: (said command)
(Response given back)
(Another said command)
(Response back)

This is practically needed for almost anything that uses cli. It is rare that you can do cli work with one line of command because most will do as such as the above example. Can this and other promises features be implemented. Many thanks.

**Reply by Arvinth-Krishna** [reply]
@BigFlubba  Thank you for your suggestion! This looks like a needed feature to be added to Reco. I will try to add it in the following days.
